File: src/main/java/com/kelsos/mbrc/commands/ProcessUserAction.java
package com.kelsos.mbrc.commands;

import com.google.inject.Inject;
import com.kelsos.mbrc.model.SocketMessage;
import com.kelsos.mbrc.model.UserAction;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.constants.Protocol;
import com.kelsos.mbrc.net.SocketService;

public class ProcessUserAction implements ICommand {
    private SocketService socket;

    @Inject public ProcessUserAction(SocketService socket) {
        this.socket = socket;
    }

    @Override public void execute(IEvent e) {
        socket.sendData(new SocketMessage(((UserAction) e.getData()).getContext(), Protocol.Request,
                ((UserAction) e.getData()).getData()));
    }
}

File: app/src/main/java/com/kelsos/mbrc/commands/ProcessUserAction.kt
package com.kelsos.mbrc.commands

import com.kelsos.mbrc.data.UserAction
import com.kelsos.mbrc.networking.client.SocketMessage
import com.kelsos.mbrc.networking.client.SocketService
import com.kelsos.mbrc.networking.protocol.ProtocolAction
import com.kelsos.mbrc.networking.protocol.ProtocolMessage

class ProcessUserAction(
  private val socket: SocketService,
) : ProtocolAction {
  override fun execute(message: ProtocolMessage) {
    socket.sendData(
      SocketMessage.create(
        (message.data as UserAction).context,
        (message.data as UserAction).data,
      ),
    )
  }
}
